### PR TITLE
Created `shape` , `dimensions`, and `volume` methods on the `Core` class

### PR DESCRIPTION
**Pull request by Atomator**
Created various methods on the `Core` class to help with determining its shape, size, and volume. The methods written for this ticket were as follows:
- `shape` - returns the shape of the `pixel_array`
- `dimensions` - returns the dimensions of the `pixel` array in mm
- `volume` - returns the approximate volume of the core in cubic mm by first counting the number of voxels within a certain density range and then multiplying that count by the volume of a single voxel  

Now it is important to note that while all of these functions work, I'm not sold on their execution—more specifically, I don't love the way that the volume function works in having to take a range of density values. I made it take a range of density values so white noise can be more easily cut out, but it is still kinda a messy way to approximate the volume. If y'all have any better ideas, let me know!

### CODE REVIEW

File: src/core_ct/core.py
Comment (TheSecondReal0) [inline src/core_ct/core.py:313]: I think this would be more readable if it was
```
size * dimension for size, dimension
in zip(self.pixel_array.shape, self.pixel_dimensions)
```

File: src/core_ct/core.py
Comment (TheSecondReal0) [inline src/core_ct/core.py:318]: I'm not sure what the standard is but this looks a little weird to me. Maybe this would be better but not sure
```
def volume(self, 
    min_density: float = 0, 
    max_density: float = float("inf")
    ) -> float:
```

File: src/core_ct/core.py
Comment (TheSecondReal0) [inline src/core_ct/core.py:317]: Also negative brightness values are valid, so the `min_density` default should be negative infinity

**Review by TheSecondReal0 — CHANGES_REQUESTED** [review]

File: src/core_ct/core.py
Comment (carlameii) [inline src/core_ct/core.py:287]: Because `pixel_dimensions` is already an attribute of the core that refers to something completely different than this, it feels confusing to say that the shape function returns the pixel dimensions of the scan. Maybe change to "get the dimensions of the pixel array of the core scan" or something similar.

File: src/core_ct/core.py
Comment (carlameii) [inline src/core_ct/core.py:324]: Can we specify these are floats in the docstring too?

File: src/core_ct/core.py
Comment (carlameii) [inline src/core_ct/core.py:309]: Can we specify this is a tuple in the docstring?

File: src/core_ct/core.py
Comment (Atomator) [reply src/core_ct/core.py:313]: I might be able to switch that around, but rn that formatting is being done automatically by the Black Python Formatter so I don't really have control over it. If you really want me to adjust it, I can just stop using Black or look into disabling it, but if you don't think it's that big of a deal I'd prefer to leave it.

File: src/core_ct/core.py
Comment (Atomator) [reply src/core_ct/core.py:318]: Same issue as in the comment I left on the `return tuple` statement. Quoting that comment below for readability:

> I might be able to switch that around, but rn that formatting is being done automatically by the Black Python Formatter so I don't really have control over it. If you really want me to adjust it, I can just stop using Black or look into disabling it, but if you don't think it's that big of a deal I'd prefer to leave it.

File: src/core_ct/core.py
Comment (Atomator) [reply src/core_ct/core.py:317]: Talked about this in person. We decided on removing the `min_density` and `max_density` arguments and making this function reliant on the filter method.

File: src/core_ct/core.py
Comment (Atomator) [reply src/core_ct/core.py]: Updated this to specify that it returns a "three-element tuple"

File: src/core_ct/core.py
Comment (Atomator) [reply src/core_ct/core.py:324]: Removed these arguments as per the comment thread that I had with @TheSecondReal0 left on the `volume` method.

> Talked about this in person. We decided on removing the min_density and max_density arguments and making this function reliant on the filter method.

**Comment by Atomator** [comment]
Since we decided to make the `volume` function reliant on the `filter` function to remove values that we don't want to be counted towards the volume, I am going to wait until the `filter` method gets merged into `main` to fix the tests for `volume` function to use the `filter` method. So if you notice those tests are failing rn, that's why

**Comment by Atomator** [comment]
Quick update. I just went ahead and removed the failing tests, but there's still a TODO comment saying that it should be updated to work with the filter method, so we just need to make sure that happens before this branch gets merged in

**Review by carlameii — APPROVED** [review]

**Review by TheSecondReal0 — APPROVED** [review]

**Pull request merged**
